radiation/python/radlw/taumol_gt4py.py

```diff
@@ -0,0 +1,4957 @@
+    stencil,
+    mod,
+)
+
+sys.path.insert(0, "/Users/AndrewP/Documents/work/physics_standalone/radiation/python")
+from config import *
+from util import create_storage_from_array, create_storage_zeros, compare_data
+from radlw.radlw_param import (
+    nrates,
+    nspa,
+    nspb,
+    ngb,
+    ng01,
+    ng02,
+    ng03,
+    ng04,
+    ng05,
+    ng06,
+    ng07,
+    ng08,
+    ng09,
+    ng10,
+    ng11,
+    ng12,
+    ng13,
+    ng14,
+    ng15,
+    ng16,
+    ns02,
+    ns03,
+    ns04,
+    ns05,
+    ns06,
+    ns07,
+    ns08,
+    ns09,
+    ns10,
+    ns11,
+    ns12,
+    ns13,
+    ns14,
+    ns15,
+    ns16,
+    oneminus,
+)
+
+SERIALBOX_DIR = "/Users/AndrewP/Documents/code/serialbox2/install"
+sys.path.append(SERIALBOX_DIR + "/python")
+import serialbox as ser
+
+rebuild = False
+validate = True
+backend = "gtc:gt:cpu_ifirst"
+
+invars = [
+    "laytrop",
+    "pavel",
+    "coldry",
+    "colamt",
+    "colbrd",
+    "wx",
+    "tauaer",
+    "rfrate",
+    "fac00",
+    "fac01",
+    "fac10",
+    "fac11",
+    "jp",
+    "jt",
+    "jt1",
+    "selffac",
+    "selffrac",
+    "indself",
+    "forfac",
+    "forfrac",
+    "indfor",
+    "minorfrac",
+    "scaleminor",
+    "scaleminorn2",
+    "indminor",
+    "nlay",
+    "fracs",
+    "tautot",
+    "NGB",
+]
+
+integervars = ["jp", "jt", "jt1", "indself", "indfor", "indminor"]
+fltvars = [
+    "pavel",
+    "coldry",
+    "colamt",
+    "colbrd",
+    "fac00",
+    "fac01",
+    "fac10",
+    "fac11",
+    "selffac",
+    "selffrac",
+    "forfac",
+    "forfrac",
+    "minorfrac",
+    "scaleminor",
+    "scaleminorn2",
+]
+
+ddir = "../../fortran/radlw/dump"
+serializer = ser.Serializer(ser.OpenModeKind.Read, ddir, "Serialized_rank0")
+
+savepoints = serializer.savepoint_list()
+
+indict = dict()
+
+for var in invars:
+    if var == "NGB":
+        indict[var] = np.tile(np.array(ngb)[None, None, :], (npts, 1, 1))
+    else:
+        tmp = serializer.read(var, serializer.savepoint["lwrad-taumol-input-000000"])
+        if var == "colamt" or var == "wx":
+            indict[var] = np.tile(tmp[None, None, :, :], (npts, 1, 1, 1))
+        elif var == "tauaer" or var == "fracs" or var == "tautot":
+            indict[var] = np.tile(tmp.T[None, None, :, :], (npts, 1, 1, 1))
+        elif var == "rfrate":
+            indict[var] = np.tile(tmp[None, None, :, :, :], (npts, 1, 1, 1, 1))
+        elif var in integervars or var in fltvars:
+            indict[var] = np.tile(tmp[None, None, :], (npts, 1, 1))
+        else:
+            indict[var] = tmp[0]
+
+indict_gt4py = dict()
+
+for var in invars:
+    if var == "NGB":
+        indict_gt4py[var] = create_storage_from_array(
+            indict[var],
+            backend,
+            shape_2D,
+            (DTYPE_INT, (ngptlw,)),
+            default_origin=(0, 0),
+        )
+    elif var == "colamt":
+        indict_gt4py[var] = create_storage_from_array(
+            indict[var], backend, shape_nlay, type_maxgas
```

Comment:
where is shape_nlay defined? this is the first occurrence of it in this file-- must come from the * import of config?

Reply:
Yes, all the shapes and types are defined in config.py. This seems to be the case for the other physics standalones too.